gnovm/pkg/gnolang/types.go

```diff
@@ -2098,6 +2108,44 @@ func assertSameTypes(lt, rt Type) {
 	}
 }
 
+func isEqualityTypes(lt, rt Type) bool {
+	debugPP.Printf("check isEqualityTypes, lt: %v, rt: %v, isLeftDataByte: %v, isRightDataByte: %v \n", lt, rt, isDataByte(lt), isDataByte(rt))
+	if lpt, ok := lt.(*PointerType); ok { // refer to 0f31
+		lt = lpt.Elt
+	}
+	if rpt, ok := rt.(*PointerType); ok {
+		rt = rpt.Elt
+	}
+	// lt or rt could be nil in runtime, e.g. a == nil, type of RHS would be nil
+	if lt == nil && rt == nil {
+		// both are nil.
+	} else if lt == nil || rt == nil {
+		// one is nil.  see function comment.
```

Comment:
there is no function comment

Reply:
same as above, this logic moved to #1890.